test/extended/images/mongodb_replica.go

```diff
@@ -0,0 +1,126 @@
+package images
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/db"
+)
+
+var _ = g.Describe("[images][mongodb] openshift mongodb replication", func() {
+	defer g.GinkgoRecover()
+
+	const (
+		templatePath         = "https://raw.githubusercontent.com/openshift/mongodb/master/2.4/examples/replica/mongodb-clustered.json"
+		deploymentConfigName = "mongodb"
+		insertCmd            = `db.bar.save({ "status": "passed" })`
+	)
+
+	const (
+		expectedReplicasAfterDeployment = 3
+		expectedReplicasAfterScalingUp  = expectedReplicasAfterDeployment + 2
+	)
+
+	oc := exutil.NewCLI("mongodb-replica", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
```

Comment:
Not sure what's the situation with other tests, but since we're using a framework such as Ginkgo, we could make the nesting and messages better?

For now we have:

```
Describe("[images][mongodb] openshift mongodb replication"
|- Describe("creating from a template"
   |- g.It( ...
      |- [ big block with all of the test ]
```

Compare with:

```
Describe("[images][mongodb] openshift mongodb replication"
|- Describe("creation from a template"
   |- g.It( ...
|- Describe("write and read data"
   |- g.It( ...
|- Describe("scale up"
   |- g.It( ...
```

The exact descriptions and sections would be up to you, @php-coder :ru: